### PR TITLE
WebGL 2 (work in progress)

### PR DESCRIPTION
**Pull request by guycalledfrank**
Don't merge yet.

### CODE REVIEW

File: src/graphics/device.js [hunk 1]
```diff
@@ -27,7 +27,11 @@ pc.extend(pc, function () {
     };
 
     var _createContext = function (canvas, options) {
+        // #ifdef WEBGL2
```
Comment (daredevildave) [inline src/graphics/device.js]: Wouldn't it be better if this was a runtime option?

Comment (Maksims) [inline src/graphics/device.js]: We had discussion about it, and did agreed that it shall be runtime option.

Comment (daredevildave) [inline src/graphics/device.js]: should be? ;-)

Comment (guycalledfrank) [inline src/graphics/device.js]: It will be. For now it's easier to keep a define.

Comment (Maksims) [inline src/graphics/device.js]: But that means it is not tested well. Changing it earlier might highlight some problems.

**Comment by Maksims** [comment]
Pretty much all features been re-implemented as separate PR's to support WebGL 2.0 new capabilities based on this PR and few extra things.
Due to that closing this PR.

Great work!

**Pull request closed without merging**